Attachments/adding-attachments.md

```diff
@@ -0,0 +1,251 @@
+## How to upload and attach files from a third party applications
+
+The basic flow of uploading files in Lime Field looks like this:
+
+1. Create a `File` entity
+2. Upload file content to that entity's endpoint
+3. Connect the file entity as attachment to another entity in the API, e.g. `Case`, `Unit`, `Contract`
+
+__Note__, you probably will need to ensure that CORS requests are allowed from the domain where your application is hosted. Contact __Lime Field__ about that.
+
+So, lets see how the uploads can be done as in a code example.
+
+### Setup and Helpers
+
+First we will define some basic varables and helper functions to make the example code more readable. Hopefully.
+
+Replace `USERNAME `, `PASSWORD` and `INSTALLATION_NAME` with valid values for your installation of Lime Field.
+
+The `entityHref` should point out the entity to which we add attachments.
+
+```javascript
+const username = 'USERNAME';
+const password = 'PASSWORD';
+const credentials = window.btoa(`${username}:${password}`);
+const baseurl = 'https://INSTALLATION_NAME.remotex.net/api';
+const entityHref = 'entity/id';
```

Comment:
If we want to make the code more reusable I think we need to be able to define `entityHref` in the upload function and not here as a constant.

Reply:
This is an example code snippet, so primary focus is on readability and ease of understanding. So, I think having all setup first is clearer.